plugins/modules/zos_copy.py

```diff
@@ -3566,10 +3588,25 @@ def run_module(module, arg_def):
                     dest_member_exists = dest_exists and data_set.DataSet.files_in_data_set_members(root_dir, dest)
                 elif src_ds_type in data_set.DataSet.MVS_PARTITIONED:
                     dest_member_exists = dest_exists and data_set.DataSet.data_set_shared_members(src, dest)
-
     except Exception as err:
         module.fail_json(msg=str(err))
-
+    identical_gdg_copy = module.params.get('identical_gdg_copy', False)
+    if identical_gdg_copy:
+        # Validate destination isn't a generation pattern
+        if is_member(dest):
```

Comment:
`raw_dest` variable contains the `dest` name without escaping, not the data set type. We could just change it to:

```suggestion
        if dest_ds_type != "GDG"
```